app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def create
+    @commentable = find_commentable
+    @comment = @commentable.comments.build(comment_params)
+  end
+
+  private
+
+  def find_commentable
+    params.each do |name, value|
+      return Regexp.last_match(1)
+        .classify
+        .constantize
+        .find(value) if name =~ /(.+)_id$/
+    end
```

Comment:
You could send the `commentable_type` in the params, that way you'll need less logic to get the right model class.